### PR TITLE
Fix #311340: Save online via command line (--save-online)

### PR DESCRIPTION
**Pull request by shoogle**
Resolves: https://musescore.org/en/node/311340

Adds `--save-online` option for command line:

```bash
mscore --save-online score1.mscz score2.mscx
```

Feature is useful for updating scores that are stored in a shared Git repository, or other location where they are editable by multiple people, but where only one person has permission to upload and publish to Musescore.com.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/WorkflowAndGuidelines/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [N/A] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
A few coding style errors, nothing really serious

**Comment by Jojo-Schmitz** [comment]
Wouldn't we need this for 3.x (3.6 actually) too?

File: mscore/musescore.cpp [hunk 1]
```diff
@@ -8083,6 +8092,7 @@ MuseScoreApplication::CommandLineParseResult MuseScoreApplication::parseCommandL
     parser.addOption(QCommandLineOption({ "E", "install-extension" },
                                         "Install an extension, load soundfont as default unless -e is passed too",
                                         "extension file"));
+    parser.addOption(QCommandLineOption("save-online", "Upload score(s) to their source URL. Replaces existing online score(s)."));
```
Comment (Jojo-Schmitz) [inline mscore/musescore.cpp:8095]: Maybe add that this is not doing 1st time uploads, just updating existing ones and even that only for the same user?

Comment (Jojo-Schmitz) [reply mscore/musescore.cpp:8095]: And please also update build/Linux+BSD/mscore.1.in

Comment (shoogle) [reply mscore/musescore.cpp:8095]: Not sure there's room here, but I added a longer explanation in the manpage.

**Review by Jojo-Schmitz — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
Are you going to port this to 3.x too?

**Comment by shoogle** [comment]
I updated the date. I will look at 3.x, but if it is very different then I won't bother. FYI, this seems to work for uploading 3.x scores.

**Comment by Jojo-Schmitz** [comment]
"Save online" 3.x score with master builds may work now, but won't as soon as master changes to a newer file format, which I believe will happen sooner or later.

**Comment by Jojo-Schmitz** [comment]
The difficult part of porting this from master to 3.x is the m4/cloud changes

**Pull request merged**